minql-uri/src/authority.rs

```diff
@@ -30,17 +32,72 @@ use crate::{hostinfo::HostInfo, userinfo::UserInfo};
 /// must be enclosed in brackets ([]).
 /// > * An optional port subcomponent preceded by a colon (:), consisting of decimal digits.  
 ///
-/// ```bnf
+/// ## ABNF Grammar
+/// ```abnf
 /// authority = [userinfo "@"] host [":" port]
 /// ```
 #[derive(Debug)]
 pub struct Authority<'str> {
     /// Raw unparsed Authority String
-    pub string: &'str str,
+    pub raw: &'str str,
     /// Authority User Information
     pub userinfo: Option<UserInfo<'str>>,
     /// Authority Host Information
     pub hostinfo: HostInfo<'str>,
     /// Authority Port Number
     pub port: Option<u16>,
 }
+
+impl<'str> Authority<'str> {
+    /// Convert Parsed Authority into a Builder
+    #[must_use]  
+    pub fn builder(&self) -> AuthorityBuilder {
+        AuthorityBuilder {
+            userinfo: self.userinfo.as_ref().map(UserInfo::builder),
+            hostinfo: self.hostinfo.builder(),
+            port: self.port,
+        }
+    }
+}
+
+impl<'str> std::fmt::Display for Authority<'str> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.raw)
+    }
+}
```

Comment:
The raw version is absolutely the version without modifications since this version should not be modified. This isn't implemented as a copy-on-write. Utilize the builder to modify and reconstruct the string.